org.mwc.debrief.core/src/org/mwc/debrief/core/wizards/dynshapes/DynamicCircleWizard.java

```diff
@@ -56,11 +56,19 @@ public void addPages()
   public boolean performFinish()
   {
     Date startTime = _shapeTimingsPage.getStartTime();

```

Comment:
@ayeshare - we've got lots of code duplication in these three blocks of code for the DynamicShapes.

Is it an option to refactor, to pull out common elements?

I'd _typically_ handle it by the core class having `abstract protected PlainShape getShape()` that gets called from the core code.